python/beeai_framework/agents/base.py

```diff
@@ -57,17 +110,24 @@ def meta(self) -> AgentMeta:
             tools=[],
         )
 
+    def run_config(self, **kwargs: Unpack[AgentOptions]) -> AgentExecutionConfig:
```

Comment:
What is it? The name is very confusing; it is called run, but it does not run anything. We don't want to have dozens of methods (especially public). 

If you wanted to address my prior comment, I was talking about adding a factory function (in utils) or factory static method to the `AgentExecutionConfig`.

Reply:
Right, agree. Let me change it.